jobbergate-api/jobbergate_api/apps/applications/routers.py

```diff
@@ -260,23 +260,36 @@ async def applications_list(
 
 
 @router.get(
-    "/applications/{application_id}",
+    "/applications/{application_identification}",
     description="Endpoint to return an application given the id",
     response_model=ApplicationResponse,
     dependencies=[Depends(guard.lockdown(Permissions.APPLICATIONS_VIEW))],
 )
-async def applications_get_by_id(application_id: int = Query(...)):
+async def applications_get_by_id(
+    application_identification: Union[int, str] = Query(None),
+):
     """
-    Return the application given it's id.
+    Return the application given it's id (when int) or identifier (when str).
     """
-    logger.debug(f"Getting {application_id=}")
+    logger.debug(f"Getting {application_identification=}")
+
+    if isinstance(application_identification, int):
+        query = applications_table.select().where(applications_table.c.id == application_identification)
+    elif isinstance(application_identification, str):
+        query = applications_table.select().where(
+            applications_table.c.application_identifier == application_identification
```

Comment:
We need to discuss the best approach to make sure that `application_identifier` is url safe

Reply:
Further tests indicate that it is possible to create applications with complex identifiers, like:

    jobbergate applications create --name test-test --application-path ../../00_smallcase/ --identifier "Hellö Wörld@Python"

And then use this identifier to create a job script:

    jobbergate job-scripts create --name test --application-identifier "Hellö Wörld@Python"
    
The identifier is converted internally, resulting in:

    jobbergate-api_1    | INFO:     172.22.0.1:38168 - "GET /jobbergate/applications/Hell%C3%B6%20W%C3%B6rld%40Python HTTP/1.1" 200 OK

Reply:
So, I'm solving the discussion